master/internal/api_master.go

```diff
@@ -25,3 +26,60 @@ func (a *apiServer) GetMaster(
 		Config:    configStruct,
 	}, err
 }
+
+// effectiveOffset Returns effective offset.
+func effectiveOffset(reqOffset int, totalItems int) (offset int) {
+	offset = reqOffset
+	if reqOffset < 0 {
+		offset = totalItems + offset
+		if offset < 0 {
+			offset = 0
+		}
+	}
+	return offset
+}
+
+// effectiveLimit Returns effective limit.
+// Input: Limit 0 is treated as no limit
+// Output: Limit -1 is treated as no limit
+func effectiveLimit(reqLimit int, offset int, totalItems int) (limit int) {
+	limit = -1
+	if reqLimit != 0 {
+		limit = reqLimit
+		if limit > totalItems-offset {
+			limit = totalItems - offset
+		}
+	}
+	return limit
+}
+
+// TODO could we have this work with generic requests that have offset and limit?
```

Comment:
@stoksc do you know if we can do this? Although I'd probably avoid changing much in this PR.

Reply:
Yeah we probably could - I guess the idea is trial logs and master logs probably have the same semantics as far as offset/limit are concerned.

Probably can leave it off this PR though, and make a ticket for it. Little things like this pile up and it wouldn't be too much work in a follow-up.

Reply:
https://determinedai.atlassian.net/browse/DET-3776